flow/layers/layer.h

```diff
@@ -44,19 +44,41 @@ enum Clip { none, hardEdge, antiAlias, antiAliasWithSaveLayer };
 
 class ContainerLayer;
 
-struct PrerollContext {
+struct RasterContext {
   RasterCache* raster_cache;
   GrContext* gr_context;
-  ExternalViewEmbedder* view_embedder;
-  MutatorsStack& mutators_stack;
   SkColorSpace* dst_color_space;
-  SkRect cull_rect;
-
-  // The following allows us to paint in the end of subtree preroll
   const Stopwatch& raster_time;
   const Stopwatch& ui_time;
   TextureRegistry& texture_registry;
   const bool checkerboard_offscreen_layers;
+};
+
+using PrerollRasterOperation =
+    std::function<void(RasterContext* raster_context)>;
+
+class PrerollRasterOperations {
```

Comment:
Lets call this `GPUPrerollOperations` or `RenderPrerollOperations` and before calling each method, DCHECK to ensure you are on the GPU task runner.